src/powerhouse/hooks/usePagination.ts

```diff
@@ -0,0 +1,135 @@
+ *   pages: PageItem[]; - The array of page items.
+ *   goToPage: (page: number) => void; - The function to go to a specific page.
+ *   pageItems: T[]; - The array of items for the current page.
+ *   pageCount: number; - The total number of pages.
+ *   currentPage: number; - The current page index.
+ *   goToLastPage: () => void; - The function to go to the last page.
+ *   goToFirstPage: () => void; - The function to go to the first page.
+ *   goToNextPage: () => void; - The function to go to the next page.
+ *   goToPreviousPage: () => void; - The function to go to the previous page.
+ *   isNextPageAvailable: boolean; - The flag to indicate if the next page is available.
+ *   isPreviousPageAvailable: boolean; - The flag to indicate if the previous page is available.
+ * }} - The pagination object with various functions and properties.
+ */
+
+export function usePagination<T>(
+    items: T[],
+    options?: UsePaginationOptions,
+): UsePaginationResult<T> {
+    const { itemsPerPage = 10, initialPage = 0, pageRange = 5 } = options || {};
+
+    const [currentPage, setCurrentPage] = useState(initialPage);
+
+    const pageCount = Math.ceil(items.length / itemsPerPage);
+    const isNextPageAvailable = currentPage < pageCount - 1;
+    const isPreviousPageAvailable = currentPage > 0;
+
+    const goToNextPage = useCallback(() => {
+        if (isNextPageAvailable) {
+            setCurrentPage(prev => prev + 1);
+        }
+    }, [isNextPageAvailable, setCurrentPage]);
+
+    const goToPreviousPage = useCallback(() => {
+        if (isPreviousPageAvailable) {
+            setCurrentPage(prev => prev - 1);
+        }
+    }, [isPreviousPageAvailable, setCurrentPage]);
+
+    const goToLastPage = useCallback(() => {
+        setCurrentPage(pageCount - 1);
+    }, [pageCount]);
+
+    const goToFirstPage = useCallback(() => {
+        setCurrentPage(0);
+    }, []);
+
+    const goToPage = useCallback(
+        (page: number) => {
+            if (page >= 0 && page < pageCount) {
+                setCurrentPage(page);
+            }
+        },
+        [pageCount],
+    );
+
+    const pageOffset = Math.floor((pageRange - 1) / 2);
+    const availableRange = Math.min(pageRange, pageCount);
+    const maxStartIndex = Math.max(pageCount - availableRange);
+    const startIndex = Math.min(
+        maxStartIndex,
+        Math.max(currentPage - pageOffset, 0),
+    );
+
+    const pages = useMemo(() => {
+        const range: PageItem[] = [];
+        for (let i = startIndex; i < availableRange + startIndex; i++) {
+            range.push({
+                index: i,
+                active: i === currentPage,
+                number: i + 1,
+            });
+        }
+        return range;
+    }, [availableRange, startIndex, currentPage]);
+
+    const pageItems = items.slice(
+        currentPage * itemsPerPage,
+        (currentPage + 1) * itemsPerPage,
+    );
+
+    const hiddenNextPages = pages.slice(-1)[0].index < pageCount - 1;
```

Comment:
```suggestion
    const hiddenNextPages = pages.slice(-1)[0].index < pageCount - 1;
```

the pages.slice array access doesn't know that the array might be empty, and therefore will return undefined here. This makes the stories with empty data break and will also break when you create a new empty document.

I'd do this instead:

```suggestion
    const lastIndex = pages.at(-1)?.index;
    const hiddenNextPages = lastIndex ? lastIndex < pageCount - 1 : false;
    ```
    
    The .at() method knows that it might return undefined, so it warns you to handle this case.

Reply:
nice catch! fixed here: [0cba9e8](https://github.com/powerhouse-inc/design-system/commit/0cba9e8b3b9c22e97d097504c9dd86d240f15d44)